### PR TITLE
Main page returns ANSI text when fetched by curl

### PR DESCRIPTION
**Pull request by 89netraM**
When running `curl https://digit.chalmers.it` the HTML of the page is printed to the console. Something fun we can do instead is to print an ANSI text representation of the page. That is what this PR does.

It's just a fun easter egg that someone might find.

### CODE REVIEW

**Review by rosengrenen — DISMISSED** [review]
pog

**Review by hulthe — DISMISSED** [review]
[7m pog [27m

File: src/main.rs [hunk 1]
```diff
@@ -13,6 +16,29 @@ use std::collections::HashMap;
...
+
+    async fn from_request(request: &'r Request<'_>) -> Outcome<Self, Self::Error> {
+        if let Some(user_agent) = request.headers().get_one("User-Agent") {
+            if user_agent.contains("curl") {
+                return Outcome::Success(TerminalOutput(true));
+            }
+        }
+
+        if let Some(media_type) = request.headers().get_one("Accept") {
+            if media_type.contains("text/plain") {
```
Comment (Deltaphish) [inline src/main.rs:33]: This makes the site unusable when using the lynx browser. Maybe make sure that the media_type does not contain the "text/html" string too?

**Review by Deltaphish — CHANGES_REQUESTED** [review]
This breaks navigation for the lynx browser. Sure the market share is probably < 0.001%, but if we don't care about supporting old, obscure tech then who would?

File: src/main.rs (hunk 1 above)
Comment (89netraM) [reply src/main.rs:33]: How does a request from the lynx browser look? What breaks?

Comment (Deltaphish) [reply src/main.rs:33]: this is the media_type: `text/html, text/plain, text/sgml, text/css, application/xhtml+xml, */*;q=0.01
`, which triggers the `.contains("text/plain")

Comment (89netraM) [reply src/main.rs:33]: Why? 🤔 I'll add the `!media_type.contains("text/html")` check and hope it'll work better.

Comment (Deltaphish) [reply src/main.rs]: yeah http headers are never consistent. Add that check and everything should work

**Review by Deltaphish — APPROVED** [review]
Works on my lynx :thumbsup:

**Review by hulthe — APPROVED** [review]
\<b\>pog\</b\>

**Pull request merged**